modules/alloy/src/main/ts/ephox/alloy/behaviour/blocking/BlockingApis.ts

```diff
@@ -0,0 +1,80 @@
+import { Optional, Optionals } from '@ephox/katamari';
+import { Attribute } from '@ephox/sugar';
+import * as BehaviourTypes from '../common/BehaviourTypes';
+import * as Behaviour from '../../api/behaviour/Behaviour';
+import { Keying } from '../../api/behaviour/Keying';
+import { Focusing } from '../../api/behaviour/Focusing';
+import { Replacing } from '../../api/behaviour/Replacing';
+import { AlloySpec } from '../../api/component/SpecTypes';
+import { AlloyComponent } from '../../api/component/ComponentApi';
+import * as GuiFactory from '../../api/component/GuiFactory';
+import { BlockingConfig } from './BlockingSchema';
+import { BlockingState } from './BlockingState';
+
+export interface BlockingApis extends BehaviourTypes.BehaviourApisRecord<BlockingConfig, BlockingState> {
+  block: typeof block;
+  unblock: typeof unblock;
+}
+
+// Mark this component as busy, or blocked.
+const block = (
+  component: AlloyComponent,
+  config: BlockingConfig,
+  state: BlockingState,
+  // This works in conjunction with the 'getRoot' function in the config. To
+  // attach a blocker component to the dom, ensure that 'getRoot' returns a
+  // component, and this function returns the specification of the component to
+  // attach.
+  getBusySpec?: (behaviour: Behaviour.AlloyBehaviourRecord) => AlloySpec
+) => {
+  if (!state.getBlocked()) {
+    config.onBlock(component);
+  }
+
+  state.setBlocked(true);
+
+  Attribute.set(component.element, 'aria-busy', true);
+
+  let hasNewBlocker = false;
```

Comment:
Instead of using a mutable variable here, you could return the value from `lift2` and then act on that. eg:
```
const newBlocker = Optionals.lift2(Optional.from(getBusySpec), config.getRoot(component), (getBusySpec, root) => {
  ...
  return blocker;
});

newBlocker.fold(state.clearBlocker, Fun.noop);
```

- [ ] Avoid the mutable variable if possible

Reply:
Fixed, albeit slightly differently